src/routes/rides.ts

```diff
@@ -15,172 +23,314 @@ router.get("/", async (_, res) => {
+    await db.insert(stops).values(rideStops.map((value, index) => {
+      return {
+        order: index + 1,
+        rideId,
+        location: value,
+      };
+    }));
+
+    // Insert the owner as a ride member
+    await db.insert(rideMembers).values({ rideId, userId });
+  } catch (_) {
+    res.status(StatusCodes.INTERNAL_SERVER_ERROR).json({
+      message: "Failed to create a ride request! Please try again later",
+    });
+    return;
+  }
+
+  res.end();
+});
+
+// Request to join a ride
+router.post("/request/:rideId", async (req, res) => {
+  const { userId } = res.locals;
+  if (!userId) return;
+
+  if (!req.params?.rideId) {
+    res.status(StatusCodes.BAD_REQUEST).json({
+      message: "No ride id provided!",
+    });
+    return;
+  }
+
+  const rideId = parseInt(req.params.rideId);
 
   try {
     // Check if ride members can allow another member into the ride
     const ride = await db.query.rides.findFirst({
       where: (rides, { eq }) => eq(rides.id, rideId),
     });
 
-    const member_count = (await db.query.rideMembers.findMany({
-      columns: { rideId: false, userId: false },
+    if (!ride) {
+      res.status(StatusCodes.NOT_FOUND).json({
+        message: "The given ride was not found!",
+      });
+      return;
+    }
+
+    const members = await db.query.rideMembers.findMany({
       where: (members, { eq }) => eq(members.rideId, rideId),
-    })).length;
+    });
+
+    // User is already in ride / they created the ride
+    if (
+      members.filter((member) => member.userId == userId).length > 0 ||
+      ride.createdBy == userId
+    ) {
+      res.status(StatusCodes.CONFLICT).json({
+        message: "The given user is already a member of the ride",
+      });
+      return;
+    }
 
     // Should this be checked for a request?
-    if (member_count >= (ride?.maxMemberCount ?? 0)) {
-      // Service unavailable
-      res.status(503).json({
+    if (members.length >= (ride?.maxMemberCount ?? 0)) {
+      res.status(StatusCodes.SERVICE_UNAVAILABLE).json({
         message:
           "The requested ride is already full! Try requesting later when space is available or try finding another similar ride.",
       });
       return;
     }
 
     // Check if request already exists
-    const exists = await db.query.userRequests.findMany({
+    const exists = await db.query.userRequests.findFirst({
       where: (requests, { eq, and }) =>
         and(eq(requests.rideId, rideId), eq(requests.userId, userId)),
     });
 
-    if (exists.length > 0) {
+    if (exists) {
       // Conflict
-      res.status(409).json({
+      res.status(StatusCodes.CONFLICT).json({
         message:
           "Ride request from this user already exists for the given ride!",
       });
       return;
     }
 
-    await db.insert(userRequests).values({ userId, rideId });
+    await db.insert(userRequests).values({ userId, rideId, status: "pending" });
```

Comment:
the check for existing request is unnecessary, can just do insert with on conflict do nothing